modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-service/service.xml

```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0"?>
+<!DOCTYPE service-builder PUBLIC "-//Liferay//DTD Service Builder 7.4.0//EN" "http://www.liferay.com/dtd/liferay-service-builder_7_4_0.dtd">
+
+<service-builder auto-namespace-tables="false" mvcc-enabled="false" package-path="com.liferay.commerce.shop.by.diagram">
+	<author>Andrea Sbarra</author>
+	<namespace>Commerce</namespace>
+	<entity local-service="true" name="CPDefinitionDiagramSetting" remote-service="true" uuid="true">
+
+		<!-- PK fields -->
+
+		<column name="CPDefinitionDiagramSettingId" primary="true" type="long" />
+
+		<!-- Audit fields -->
+
+		<column name="companyId" type="long" />
+		<column name="userId" type="long" />
+		<column name="userName" type="String" />
+		<column name="createDate" type="Date" />
+		<column name="modifiedDate" type="Date" />
+
+		<!-- Other fields -->
+
+		<column name="CPDefinitionId" type="long" />
+		<column name="radius" type="double" />
+		<column name="color" type="String" />
+
+		<!-- Finder methods -->
+
+		<finder name="CPDefinitionId" return-type="CPDefinitionDiagramSettings" unique="true">
+			<finder-column name="CPDefinitionId" />
+		</finder>
+	</entity>
+	<entity local-service="true" name="CPDefinitionDiagramEntry" remote-service="true">
+
+		<!-- PK fields -->
+
+		<column name="CPDefinitionDiagramEntryId" primary="true" type="long" />
+
+		<!-- Audit fields -->
+
+		<column name="companyId" type="long" />
+		<column name="userId" type="long" />
+		<column name="userName" type="String" />
+		<column name="createDate" type="Date" />
+		<column name="modifiedDate" type="Date" />
+
+		<!-- Other fields -->
+
+		<column name="CPInstanceUuid" type="String" />
```

Comment:
use className classPK?

Reply:
Add Flag that say if link is SKU or Diagram

Add product code for product that is not present at catalog level